### DISCUSSION TITLE
AutoEnsemble raising ValueError: y must have two or more variables, but found only one

### DISCUSSION THREAD

**Discussion by ManuB68** [alan-turing-institute/sktime#5876] [General]
Hi everyone,

I try to use AutoEnsemble but it raises ValueError: y must have two or more variables, but found only one while my dataframee have multiple variables. It doesn't work even if I had or remove variables, remove the multi-index, different forecasters, etc. while Ensemble works perfectly with the same data frame and forecasters.

Here is some code. The data frame is small, but I have the same issue with a much bigger data frame:

```python
# use case for AutoEnsemble for SK-Time

df = pd.DataFrame([['A', '2024-01-01', 3, 13], ['A', '2024-01-02', 2, 12], ['A', '2024-01-03', 3, 13], ['A', '2024-01-04', 4, 14],\
                   ['A', '2024-01-05', 6, 14], ['A', '2024-01-06', 8, 15], ['A', '2024-01-07', 9, 16],\
                  ['B', '2024-01-01', 101, 3], ['B', '2024-01-02', 102, 2], ['B', '2024-01-03', 103, 3], ['B', '2024-01-04', 104, 4],\
                  ['B', '2024-01-05', 106, 7], ['B', '2024-01-06', 108, 6], ['B', '2024-01-07', 109, 6],\
                  ['C', '2024-01-01', 10000, 10], ['C', '2024-01-02', 20000, 20], ['C', '2024-01-03', 30000, 30], ['C', '2024-01-04', 40000, 43],\
                  ['C', '2024-01-05', 50006, 42], ['C', '2024-01-06', 49999, 66], ['C', '2024-01-07', 55555, 65]],\
                  columns=['Patients', 'Dates', 'M1', 'M2'])

# Assign date type to 'Dates'
df['Dates']= pd.to_datetime(df['Dates'])


# setting the multiindex
df = df.reset_index(drop = True)
df = df.set_index(["Patients", "Dates"])

print("df",df)


# splitting the data
y_train, y_test = temporal_train_test_split(df, test_size=1)

# specifying forecasting horizon
fh = 1

# specifying the forecasting algorithm
 
regressor = KNeighborsRegressor(n_neighbors=1)
forecaster_KNN = make_reduction(regressor, window_length=3)

forecasters = [
    ("KNN", forecaster_KNN),
    ("VAR", VAR())
]

#    ("VECM", VECM())
#   ("ETS", AutoETS(auto=True, n_jobs=-1)),
#    ("KNN160/459", forecaster_KNN)

forecaster = AutoEnsembleForecaster(forecasters=forecasters)        # works without Auto

# fitting the forecaster
forecaster.fit(y_train)

# raises the error

# querying predictions
y_pred = forecaster.predict(fh=fh)
```

**Comment by fkiraly** [comment]
@ManuB68, thanks for pointing out the issue, the error message can be improved here.

The problem arises from a combination of two things:

* `VAR` requires strictly multivariate endogenous data, that is, two or more variables. Your dataset has two (M1, M2), but not the data that `VAR` sees.
* That is because you wrap it in `AutoEnsembleForecaster`, which is an algorithm that can deal only with univariate endogenous data `y`. So, due to automatic broadcasting, one `AutoEnsmbleForecaster` is fit for `M1` and one for `M2`, and once that univariate data hits `VAR`, it causes the error.

Now, there's nothing that would keep one from generalizing the `AutoEnsembleForecaster` to multivariate data - for this, the `regressor` would have to be applied to multivariate internal `y`, so possibly wrapped in `MultiOutput` of `sklearn`, or otherwise looped over.

That contribution would be nice and appreciated.

**Reply by ManuB68** [reply]
Ok got it

@fkiraly Is EnsembleForecaster also spitting the tasks in multiple univariate models or is it truly MultiVariate ?

btw I am unable to implement these changes, sorry

**Reply by fkiraly** [reply]
You can check whether an estimator is univariate or multivariate by inspecting the `scitype:y` tag, e.g., `my_estimator.get_tag("scitype:y")` (or `get_class_tag` if you are looking at the class)

According to that, `EnsembleForecaster` is genuinely multivariate.

> btw I am unable to implement these changes, sorry

But maybe someone else will, reading this. If you would like to encourage others to pick this up, you might also want to open an issue with a feature request.
